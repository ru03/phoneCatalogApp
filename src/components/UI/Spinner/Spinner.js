import React from 'react';
import classes from './Spinner.module.css';
import Backdrop from './../Backdrop/Backdrop';
const spinner = (props) => (
    <>
        <Backdrop></Backdrop>
        <div className={classes.spinner}>
            <div className={classes.doubleBounce1}></div>
            <div className={classes.doubleBounce2}></div>
        </div>
    </>
)

export default spinner;