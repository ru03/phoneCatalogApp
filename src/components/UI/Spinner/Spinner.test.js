import React from 'react';
import { mount } from 'enzyme';
import Spinner from './Spinner';
import Backdrop from './../Backdrop/Backdrop';

describe('Spinner test', () => {
    let spinner;
    beforeEach(() => {
        spinner = mount(<Spinner />);
    });

    it('Spinner component should be displayed', () => {
        expect(spinner).toMatchSnapshot();
    });

    it('Spinner component should have a backdrop component', () => {
        expect(spinner.find(Backdrop)).toHaveLength(1);
    });
})