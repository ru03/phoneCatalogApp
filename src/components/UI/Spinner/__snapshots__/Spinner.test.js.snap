// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner test Spinner component should be displayed 1`] = `
<spinner>
  <backdrop>
    <div
      className="Backdrop"
    />
  </backdrop>
  <div
    className="spinner"
  >
    <div
      className="doubleBounce1"
    />
    <div
      className="doubleBounce2"
    />
  </div>
</spinner>
`;
